File: sistema veterinario/RegistroMedico.java
public class RegistroMedico
{
    private int codigo;
    private Animal animal;

    public RegistroMedico(int codigo,
                          Animal animal)
    {
        this.codigo = codigo;
        this.animal = animal;
    }

    public void mostrarRegistro()
    {
        System.out.println("Registro: " + codigo);
        System.out.println("Paciente: "
                           + animal.getNombre());
    }
}

File: Logica/RegistroMedico.cs
RegistroMedico.cs                                       namespace Sistema_Veterinario.Logica
{
    public class RegistroMedico
    {
        public int Codigo { get; set; }

        public RegistroMedico(int codigo)
        {
            Codigo = codigo;
        }

        public void Mostrar()
        {
            Console.WriteLine(
                "Código Registro: " + Codigo);
        }
    }
}